test/amperity/gocd/secret/vault/plugin_test.clj

```diff
@@ -0,0 +1,214 @@
+      (is (response-equal (DefaultGoPluginApiResponse/success "\"\"")
+                          (plugin/handler (mock-client-atom) (default-go-plugin-api-request nil)))))))
+
+
+(deftest handler-with-plugin-response
+  (testing "Handler when handle-method returns a GoPluginApiResponse response"
+    (with-redefs [plugin/handle-request
+                  (fn [_ _ _] {:response-code 200 :response-body {:message "hello"} :response-headers {}})]
+      (is (response-equal (DefaultGoPluginApiResponse/success "{\"message\":\"hello\"}")
+                          (plugin/handler (mock-client-atom) (default-go-plugin-api-request nil)))))))
+
+
+(deftest handler-with-json-response
+  (testing "Handler when handle-method returns a json response"
+    (let [response {:response-code 200 :response-headers {} :response-body {:try "this"}}]
+      (with-redefs [plugin/handle-request (fn [_ _ _] response)]
+        (is (response-equal (DefaultGoPluginApiResponse/success "{\"try\":\"this\"}")
+                            (plugin/handler (mock-client-atom) (default-go-plugin-api-request nil))))))))
+
+;; Endpoint Tests ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(deftest get-icon
+  (testing "Get icon endpoint with well formed requests"
+    (let [result (plugin/handle-request (mock-client-atom) "cd.go.secrets.get-icon" "")
+          body (:response-body result)
+          status (:response-code result)]
+      (is "image/svg+xml"
+          (:content_type body))
+      (is (:data body))
+      (is (= 200 status)))))
+
+
+(deftest validate
+  (with-redefs [vault/authenticate! (fn [_ _ _] true)]
+    (testing "Validate correctly handles case with no errors (no false positives)"
+      (let [result (plugin/handle-request
+                     (mock-client-atom) "go.cd.secrets.validate"
+                     {:vault_addr "https://amperity.com"
+                      :auth_method "token"
+                      :vault_token "abc123"})
+            body (:response-body result)
+            status (:response-code result)]
+        (is (= [] body))
+        (is (= 200 status))))
+    (testing "Validate correctly handles case with errors (no false negatives, no false positives)"
+      (let [result (plugin/handle-request
+                     (mock-client-atom) "go.cd.secrets.validate"
+                     {:vault_addr "protocol://amperity.com"})
+            body (:response-body result)
+            status (:response-code result)]
+        (is (= [{:key     :vault_addr
+                 :message "Vault URL must start with http:// or https://"}
+                {:key     :auth_method
+                 :message "Authentication Method is required"}]
+               body))
+        (is (= 200 status))))
+    (testing "Validate also resets the vault client if a new URL is specified, when input is valid only"
+      (with-redefs [plugin/authenticate-client-from-inputs!
+                    (fn [_ i1]
```

Comment:
Both this being short (not descriptive) and using `1` (which aliases too close to `l`) is discouraged.